README.md

```diff
@@ -34,54 +47,76 @@ Any error thrown in the function with this signature is getting returned to the
 
 ## Generic Functions
 
-For additional generic functionality like request validation or authorization functions could be defined. The functions need to have the following structure:
+The middleware supports the integration of generic functions like request validation or authorization. 
+These functions must comply with the 'AzureFunction' type from the '@azure/functions' package. 
+They are crucial for extending the middleware's capabilities while adhering to Azure's function signature requirements.
 
 ```typescript
-export type MiddlewareFunction = (context: Context, request: HttpRequest) => Promise<void>;
-```
+import { AzureFunction } from '@azure/functions';
 
-The function receives the Azure Function context and request to operate. The function needs to be passed when the middleware is configured. 
+// 'AzureFunction' type signature
+export type AzureFunction = (context: Context, ...args: any[]) => Promise<any> | void;
 
-```typescript
-export default middleware(functionHandler, [validation(schema)]);
+// Configuring middleware with generic functions
+export default middleware([validation(schema)], functionHandler, []);
 ```
 
-In the above example a `validation` function is passed with a schema. All passed functions are called before the in the defined order before the handler function containing the logic for the request is called.
+Such generic functions are executed in sequence before the main handler function. 
+If a post-execution function is necessary, it can be included in the postExecution array, the third argument in the middleware function.
```

Comment:
```suggestion
If a post-execution function is necessary, it can be included in the postExecution array, the third argument in the middleware function. The post execution functions are also executed in sequence
```
right ?